sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -477,7 +476,14 @@ protected final void removeTemporaryFiles(List<String> knownFiles, PipelineOptio
       // (produced by successfully completed bundles).
       // This may still fail to remove temporary outputs of some failed bundles, but at least
       // the common case (where all bundles succeed) is guaranteed to be fully addressed.
-      Collection<String> matches = factory.match(factory.resolve(tempDir, "*"));
+      Set<String> matches = new HashSet<>();
+      // TODO: Windows OS cannot resolves and matches '*' in the path,
+      // ignore the exception for now to avoid failing the pipeline.
+      try {
+        matches.addAll(factory.match(factory.resolve(tempDir, "*")));
```

Comment:
Right, I see few tests are failing for this reason. I am looking at them.
However, WordCount fails because of this. it is a release blocker. I think we should fix this regardless.